src/main/resources/assets/react/interfaces/useSearchInterface.ts

```diff
@@ -73,126 +48,59 @@ export function useSearchState({
-					fields: ['total']
-				}))
-			})
-				.then(response => response.json())
-				.then(json => {
-					// console.debug('json', json);
-					setInterfaceCollectionCount(json.data.queryCollections.total);
-				});
-		} else {
-			fetch(`${servicesBaseUrl}/graphQL`, {
-				method: 'POST',
-				headers: {
-					'Content-Type':	'application/json'
-				},
-				body: JSON.stringify(gql.query({
-					operation: 'queryInterfaces',
-					fields: [{
-						hits: [
-							'_name',
-							'collectionIds'
-						]
-					}]
-				}))
-			})
-				.then(response => response.json())
-				.then(json => {
-					// console.debug('json', json);
-					const {hits} = json.data.queryInterfaces
-					let found = false;
-					for (let i = 0; i < hits.length; i++) {
-						const {
-							_name,
-							collectionIds
-						} = hits[i];
-						if (_name === interfaceNameProp) {
-							found = true;
-							setInterfaceCollectionCount(collectionIds.length);
-						}
-					} // for
-					if (!found) {
-						setInterfaceCollectionCount(0);
-					}
-				});
-		}
-	}, [
-		interfaceNameProp,
-		servicesBaseUrl
-	]);
+	const [searchedStringState, setSearchedStringState] = React.useState('');
+	// console.debug('Search searchString', searchString);
 
-	const getInterfaceDocumentCount = React.useCallback(() => {
-		fetch(`./explorer/api/v1/interface/${interfaceNameProp}`, {
-			method: 'POST',
-			headers: {
-				'Content-Type':	'application/json'
-			},
-			body: JSON.stringify(gql.query({
-				operation: 'search',
-				variables: {
-					count: {
-						value: 0
-					},
-					// query: {
-					// 	required: false,
-					// 	type: 'QueryDSL',
-					// 	value: {
-					// 		matchAll: {}
-					// 	}
-					// }
-				},
-				fields: ['total']
-			}))
-		})
-			.then(response => response.json())
-			.then(json => {
-				// console.debug('json', json);
-				setInterfaceDocumentCount(json.data.search.total);
-			});
-	}, [
-		interfaceNameProp
-	]);
-
-	function search(ss :string) {
-		setSearchedString('');
-		if(!ss) {
-			setResult({
+	function searchFunction({
+		searchString,
+		start = 0
+	}: {
+		searchString: string
+		start: number
+	}) {
+		setSearchedStringState('');
+		if(!searchString) {
+			setResultState({
 				count: 0,
 				hits: [],
 				locales: [],
 				profiling: [],
 				synonyms: [],
 				total: 0
 			});
-			return;
+			setBottomBarVisible(false);
+			return new Promise((resolve/*, reject*/) => {
+				resolve(undefined);
+			});
 		}
 		/*const cachedResult = getIn(
 			cache,
-			`${interfaceName}.${ss}`
+			`${interfaceName}.${searchString}`
 		) as SearchResult;
 		if (cachedResult) {
 			setResult(cachedResult);
 			return;
 		}*/
+		setBottomBarMessageHeader('Search');
+		setBottomBarMessage(`Searching for ${searchString}`);
+		setBottomBarVisible(true);
 		setLoading(true);
-		const uri = `./explorer/api/v1/interface/${interfaceNameProp}`;
-		//console.debug(uri);
+		const uri = `${basename}/api/v1/interface/${interfaceName}`;
```

Comment:
Should probably move this to consts... it's hardcoded in at least 3 different classes.

Reply:
Those will disappear when there is only one url